packages/protocol/contracts/governance/Election.sol

```diff
@@ -1043,6 +1121,25 @@ contract Election is
     return res;
   }
 
+  /**
+   * @notice Allows to turn on/off voting over maxNumGroupsVotedFor.
+   * Once this is turned on and account voted for more than maxNumGroupsVotedFor,
+   * it is account's obligation to run updateTotalVotesByAccountForGroup, updateTotalVotesByAccount
+   * once a day.
+   * @param flag The on/off flag.
+   */
+  function setAllowedToVoteOverMaxNumberOfGroups(bool flag) public {
```

Comment:
Thinking about this and looking through the code, I think we need to *not* allow Validators and Validator Groups themselves to use the over max strategy: in `Election.forceDecrementVotes` (called by `LockedGold.slash`) we might need to iterate over voted groups to be able to slash enough CELO. A Validator could avoid getting slashed by voting for a bunch of groups with very small amounts of CELO, causing the slash function to always run out of gas.

So this function needs a check that the account is not a registered validator/group, and the group registration functions need a check that they don't have this flag set.